src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTestHelpers.cs

```diff
@@ -0,0 +1,473 @@
+
+                callback((mldsa, password, pbeParameters) =>
+                    DecodePem(mldsa.ExportEncryptedPkcs8PrivateKeyPem(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(password)), pbeParameters)));
+            }
+
+            static byte[] DecodePem(string pem)
+            {
+                PemFields fields = PemEncoding.Find(pem.AsSpan());
+                Assert.Equal(Index.FromStart(0), fields.Location.Start);
+                Assert.Equal(Index.FromStart(pem.Length), fields.Location.End);
+                Assert.Equal("ENCRYPTED PRIVATE KEY", pem.AsSpan()[fields.Label].ToString());
+                return Convert.FromBase64String(pem.AsSpan()[fields.Base64Data].ToString());
+            }
+        }
+
+        internal delegate string ExportToPemCallback(MLDsa mldsa, string password, PbeParameters pbeParameters);
+        internal static void AssertExportToEncryptedPem(
+            Action<ExportToPemCallback> callback,
+            EncryptionPasswordType passwordTypesToTest = EncryptionPasswordType.All)
+        {
+            if ((passwordTypesToTest & EncryptionPasswordType.Char) != 0)
+            {
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKeyPem(password, pbeParameters));
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKeyPem(password.AsSpan(), pbeParameters));
+            }
+
+            if ((passwordTypesToTest & EncryptionPasswordType.Byte) != 0)
+            {
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKeyPem(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(password)), pbeParameters));
+            }
+        }
+
+        internal static void AssertExportToPrivateKeyPem(Action<Func<MLDsa, string>> callback) =>
+            callback(mldsa => mldsa.ExportPkcs8PrivateKeyPem());
+
+        internal static void AssertExportToPublicKeyPem(Action<Func<MLDsa, string>> callback) =>
+            callback(mldsa => mldsa.ExportSubjectPublicKeyInfoPem());
+
+        internal delegate bool TryExportFunc(Span<byte> destination, out int bytesWritten);
+        internal static byte[] DoTryUntilDone(TryExportFunc func)
+        {
+            byte[] buffer = new byte[512];
+            int written;
+
+            while (!func(buffer, out written))
+            {
+                Array.Resize(ref buffer, buffer.Length * 2);
+            }
+
+            return buffer.AsSpan(0, written).ToArray();
+        }
+
+        internal static string? AlgorithmToOid(MLDsaAlgorithm algorithm)
+        {
+            return algorithm?.Name switch
+            {
+                "ML-DSA-44" => "2.16.840.1.101.3.4.3.17",
+                "ML-DSA-65" => "2.16.840.1.101.3.4.3.18",
+                "ML-DSA-87" => "2.16.840.1.101.3.4.3.19",
+                _ => null,
```

Comment:
Everywhere that I saw calling this either passed in a well-known value (non-null return) or threw an exception using coalesce/expression-throw.

Is there a useful reason to have the null here, or should it just throw from here?

(From the product perspective it's "the thing we throw might be different based on context".  For the tests, it's "this throwing means we need to edit this function")